subgraph/src/datasources/FortaStaking.ts

```diff
@@ -0,0 +1,248 @@
+import { Address, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import { logStore, newMockEvent } from "matchstick-as";
+import {
+  StakeDeposited as StakeDepositedEvent,
+  Rewarded as RewardedEvent,
+  Slashed as SlashedEvent,
+  Froze as FrozeEvent,
+  FortaStaking as FortaStakingContract,
+} from "../../generated/FortaStaking/FortaStaking";
+import { Reward, Slash, Stake, Staker, Subject } from "../../generated/schema";
+import { fetchAccount } from "../fetch/account";
+
+import eventId from "../utils/event";
+
+export function handleStakeDeposited(event: StakeDepositedEvent): void {
+  let subject = Subject.load(event.params.subject.toHex());
+
+  let staker = Staker.load(event.params.account.toHex());
+  let stake = Stake.load(eventId(event));
+  const fortaStaking = FortaStakingContract.bind(event.address);
+
+  if (!subject) {
+    subject = new Subject(event.params.subject.toHex());
+    subject.activeStake = fortaStaking.activeStakeFor(
```

Comment:
reading from the blockchain is slow. Is there no way to get that for the events, or to reconstruct it incrementally?

Reply:
No way to get it from event sadly